src/Corax/IndexSearcher.cs

```diff
@@ -581,6 +636,189 @@ public MultiTermMatch ContainsQuery<TScoreFunction>(string field, string contain
                     this, new ContainsTermProvider(this, _transaction.Allocator, terms, field, containsTerm), scoreFunction));
         }
 
+        public IQueryMatch SearchQuery(string field, string searchTerm, SearchOperator @operator, int analyzerId)
+        {
+             ReadOnlySpan<byte> term = Encoding.UTF8.GetBytes(searchTerm).AsSpan();
+            
+            var analyzer = _analyzers[analyzerId] ?? throw new InvalidDataException($"{nameof(SearchQuery)} requires analyzer.");
+            analyzer.GetOutputBuffersSize(searchTerm.Length, out var outputSize, out var tokenSize);
+            
+            var buffer = QueryContext.MatchesPool.Rent(outputSize + Unsafe.SizeOf<Token>() * tokenSize);
+            Span<byte> encodeBufferOriginal = buffer.AsSpan().Slice(0, outputSize);
+            Span<Token> tokensBufferOriginal = MemoryMarshal.Cast<byte, Token>(buffer.AsSpan().Slice(outputSize));
+
+            var encoded = encodeBufferOriginal;
+            var tokens = tokensBufferOriginal;
+
+            analyzer.Execute(term, ref encoded, ref tokens);
+            IQueryMatch match = null;
+            foreach (var token in tokens)
+            {
+                var word = encoded.Slice(token.Offset, (int)token.Length);
+                SearchMatchOptions mode = SearchMatchOptions.Exact;
+                if (searchTerm[0] is '*') 
+                    mode |= SearchMatchOptions.EndsWith;
+                if (searchTerm[^1] is '*')
+                    mode |= SearchMatchOptions.StartsWith;
```

Comment:
What if we have `'foo*bar'` ?

Reply:
We will perform `exact`. Currently, we allow having wildcards as prefix or suffix only.
( https://ravendb.net/docs/article-page/5.3/Csharp/indexes/querying/searching#using-wildcards )